### PR TITLE
Change custom day start to ask for confirmation and prevent immediate cron

### PR DESCRIPTION
**Pull request by crookedneighbor**
Includes commits from #5640
## Problem

Adjusting the custom day start could cause cron to run immediately.
## Solution

Ask user to confirm selection and display the next time cron will run. 

Under the hood, this sets the user's last cron to that moment, so it shouldn't trigger cron to run immediately.

This also solves the bug where a user crons (say at midnight) and then is able to set their cds to 1am and cron again at 1.

### CODE REVIEW

**Comment by crookedneighbor** [comment]
The text in the confirmation modal will need to be adjusted and the warning text (and wiki articles) will need to be changed to show the new behavior.

Also, maybe it would be more appropriate to include the warning text in the confirmation modal?

@alys ^

**Comment by crookedneighbor** [comment]
screen shot 2015-09-05 at 3 45 22 pm

**Comment by crookedneighbor** [comment]
Also, why is the selection not a drop down instead of an input box :frowning: ?

File: website/views/shared/modals/settings/change-day-start.jade [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.text-center
+  .modal-header
+    h3=env.t('changeCustomDayStart')
+
+  .modal-body
+    .alert.alert-danger
+      p=env.t('nextCron', {time: '{{nextCron | date: "M/d/yy \'@\' h:mm a"}}'})
```
Comment (Alys) [inline website/views/shared/modals/settings/change-day-start.jade]: This should use the user's preferred date format.

Comment (crookedneighbor) [inline website/views/shared/modals/settings/change-day-start.jade]: Good call.

File: common/locales/en/settings.json [hunk 2]
```diff
@@ -39,6 +39,9 @@
     "xml": "(XML)",
     "json": "(JSON)",
     "customDayStart": "Custom Day Start",
+    "changeCustomDayStart": "Change Custom Day Start?",
+    "sureChangeCustomDayStart": "Are you sure you want to change your custom day start?",
+    "nextCron": "WARNING: Your dailies will reset next on <%= time %>.",
```
Comment (Alys) [inline common/locales/en/settings.json]: Cron doesn't run exactly at the CDS (unless you're using Habitica at that time), so the wording should reflect this. Maybe something like "Your Dailies will next reset the first time you use Habitica after <%= time %>."

**Comment by lemoness** [comment]
I think the text of the confirmation modal is fine.

However, the warning text definitely needs to be changed. Are there any parts of the following paragraphs that still apply, or can we simply remove them?

"Complete all your Dailies before changing the Custom Day Start or Rest in the Inn that day. Changing your Custom Day Start may cause Cron to run immediately, but after the first day it works as expected.

Allow a window of two hours for the change to take effect. For example, if it is currently set to 0 (midnight), change it before 10pm; if you want to set it to 9pm, change it before 7pm."

**Comment by Alys** [comment]
I'm not certain the red warning formatting is needed. With the display of the next time that cron will run and the ability to back out, this is now a safe operation. It's a big improvement over the current situation, but the warning formatting makes it look more scary. :)

Regarding the current warning text, which is:

```
Complete all your Dailies before changing the Custom Day Start 
or Rest in the Inn that day. Changing your Custom Day Start may
cause Cron to run immediately, but after the first day it works as expected.

Allow a window of two hours for the change to take effect. For example,
if it is currently set to 0 (midnight), change it before 10pm; if you want to
set it to 9pm, change it before 7pm.
```

I think all of that could be deleted now - is that right Blade?

It's probably worth including in the confirmation screen text like, "Make sure you have completed your Dailies before this time!"

**Comment by Alys** [comment]
Oh, sorry Lemoness, I hadn't reloaded before I typed that.

**Comment by crookedneighbor** [comment]
How's this?

screen shot 2015-09-06 at 10 15 31 am

**Pull request merged**